compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java

```diff
@@ -51,6 +51,12 @@ private CompilerPluginManager(PackageCompilation compilation,
     }
 
     static CompilerPluginManager from(PackageCompilation compilation) {
+        // Skip initialization if the compiler plugins are already initialized for the project
+        if (!compilation.packageContext().project().compilerPluginContexts().isEmpty()) {
+            List<CompilerPluginContextIml> compilerPluginContexts =
+                    compilation.packageContext().project().compilerPluginContexts();
```

Comment:
@azinneera  Shall we try to move this `compilerPluginContexts` from `Project` to `PackageContext` something like `PackageResolution`. I feel that's more relevant. WDYT?

Reply:
+1

Reply:
Shall we create a separate issue for this?

Reply:
sure will do..